src/app/components/Account/AccountLink.tsx

```diff
@@ -1,30 +1,107 @@
-import { FC } from 'react'
+import { FC, ReactNode } from 'react'
 import { Link as RouterLink } from 'react-router-dom'
 import { useScreenSize } from '../../hooks/useScreensize'
 import Link from '@mui/material/Link'
-import { TrimLinkLabel } from '../TrimLinkLabel'
 import { RouteUtils } from '../../utils/route-utils'
+import InfoIcon from '@mui/icons-material/Info'
 import Typography from '@mui/material/Typography'
 import { SearchScope } from '../../../types/searchScope'
+import { trimLongString } from '../../utils/trimLongString'
+import { MaybeWithTooltip } from '../AdaptiveTrimmer/MaybeWithTooltip'
+import { AdaptiveTrimmer } from '../AdaptiveTrimmer/AdaptiveTrimmer'
+
+const WithTypographyAndLink: FC<{
+  to: string
+  mobile?: boolean
+  children: ReactNode
+  labelOnly?: boolean
+}> = ({ children, to, mobile, labelOnly }) => {
+  return (
+    <Typography
+      variant="mono"
+      component="span"
+      sx={{
+        ...(mobile
+          ? {
+              maxWidth: '100%',
+              overflowX: 'hidden',
+            }
+          : {}),
+      }}
+    >
+      {labelOnly ? (
+        children
+      ) : (
+        <Link component={RouterLink} to={to}>
+          {children}
+        </Link>
+      )}
+    </Typography>
+  )
+}
 
 interface Props {
   scope: SearchScope
   address: string
+
+  /**
+   * Should we always trim the text to a short line?
+   */
   alwaysTrim?: boolean
+
+  /**
+   * Any extra tooltips to display
+   *
+   * (Besides the content necessary because of potential shortening)
+   */
+  extraTooltip?: ReactNode
+
+  /**
+   * Should we display this as a simple label, with no link?
+   *
+   * (Used for own address)
+   */
+  labelOnly?: boolean
 }
 
-export const AccountLink: FC<Props> = ({ scope, address, alwaysTrim }) => {
+export const AccountLink: FC<Props> = ({ scope, address, alwaysTrim, extraTooltip, labelOnly }) => {
```

Comment:
The problem with the `from` and `to` fields is that the auto-trimming only works if  no overflow / scrolling is happening in the DOM _above_ the component that is displaying the address. If the overflow is happening in a higher level, then we can't detect it in the adaptive trimmer.

The way to ensure this is to go through the dom level by level, and boxing everything with `max-width` or similar CSS features. However, here we are inside a `table`, which doesn't support `max-width`, as far as I know.

On the short term, I added a switch called `alwaysTrimOnTable`, which we can set when using the `AccountLink` inside a table (or similar environments) to disable the adaptive behavior, and fall back to fixed length. 

This gets us: 
![image](https://github.com/oasisprotocol/explorer/assets/2093792/b4c93e24-0f03-4feb-b39a-49bc4fb2b7a9)

Trimming too much is still better then hiding the stuff in an overflow, I guess.

This could be improved, but I would rather return to this after the new event-related stuff has landed.

Reply:
> * "Emitting contract" is misbehaving a bit

That one, we can solve easily.

![image](https://github.com/oasisprotocol/explorer/assets/2093792/c04c14ca-8673-46b1-8636-6133f7504f7c)